File: lightning-boot-starters/lightning-boot-starter-generic-crud-service/src/main/java/com/jianyue/lightning/boot/starter/generic/crud/service/support/query/TransactionalQuerySupport.java
package com.jianyue.lightning.boot.starter.generic.crud.service.support.query;

import org.jetbrains.annotations.Nullable;

/**
 * 支持事务性操作的ParamSupport !!!!
 */
public interface TransactionalQuerySupport extends QuerySupport {

    /**
     * 获取一个事务定义,如果有,否则使用当前事务 或者空 !!!
     * 这里使用Object的原因是,不引入 事务相关的依赖 !!!
     * 但是实际上是一个真实的事务对象或者为空
     */
    @Nullable
    Object getTransactionDefinition();
}

File: lightning-boot-starters/lightning-boot-starter-generic-crud-service/src/main/java/com/jianyue/lightning/boot/starter/generic/crud/service/support/query/TransactionalQuerySupport.kt
package com.jianyue.lightning.boot.starter.generic.crud.service.support.query

import org.springframework.transaction.TransactionDefinition
/**
 * @author FLJ
 * @date 2023/3/2
 * @time 10:53
 * @Description 支持事务的查询定义支持 !!!
 */
interface TransactionalQuerySupport<T>: Query<T> {
    /**
     * 获取一个事务定义,如果有,否则使用当前事务 或者空 !!!
     */
    fun getTransactionDefinition(): TransactionDefinition?
}